iroh/src/node.rs

```diff
@@ -392,27 +475,15 @@ where
                 },
                 // handle incoming p2p connections
                 Some(mut connecting) = server.accept() => {
-
                     let alpn = match get_alpn(&mut connecting).await {
                         Ok(alpn) => alpn,
                         Err(err) => {
                             tracing::error!("invalid handshake: {:?}", err);
                             continue;
                         }
                     };
-                    if alpn.as_bytes() == iroh_bytes::protocol::ALPN.as_ref() {
-                        let db = handler.inner.db.clone();
-                        let custom_get_handler = custom_get_handler.clone();
-                        let auth_handler = auth_handler.clone();
-                        let collection_parser = collection_parser.clone();
-                        let rt2 = rt.clone();
-                        let callbacks = callbacks.clone();
-                        rt.main().spawn(iroh_bytes::provider::handle_connection(connecting, db, callbacks, collection_parser, custom_get_handler, auth_handler, rt2));
-                    } else {
-                        tracing::error!("unknown protocol: {}", alpn);
-                        continue;
-                    }
-                }
+                    rt.main().spawn(handle_connection(connecting, alpn, handler.inner.clone(), gossip.clone(), collection_parser.clone(), custom_get_handler.clone(), auth_handler.clone()));
```

Comment:
the error from the spawn is never handled, should at least print it

Reply:
logging the error now.